### ISSUE TITLE
virtcol() assumes extra bytes belong to next column (Vim patch 8.0.0250)

### ISSUE THREAD

**Issue by blueyed** [neovim/neovim#6269]
- `nvim --version`: v0.2.0-493-g7e7f01a3
- Vim (version: 8.0.427) behaves differently!
- Operating system/version: Arch Linux

### Actual behaviour

`virtcol()` seems to wrap over to the next char too early in Neovim.

TEST CASE:
```vim
scriptencoding utf8
new
let v:errors = []
call setline(1, '…foo')
call assert_equal(1, virtcol([1, 1]))
call assert_equal(1, virtcol([1, 2]))
call assert_equal(1, virtcol([1, 3]))
call assert_equal(2, virtcol([1, 4]))
bd!

if len(v:errors)
  throw join(v:errors, '\n')
endif
```

### Expected behaviour

Should not throw errors.

line 6: Expected 1 but got 2
line 7: Expected 1 but got 2

### Steps to reproduce using `nvim -u NORC`

```
nvim -u script-from-above
```

> virtcol({expr})						*virtcol()*
		The result is a Number, which is the screen column of the file
		position given with {expr}.  That is, the last screen position
		occupied by the character at that position, when the screen
		would be of unlimited width.  …

**Comment by blueyed** [comment]
Vim 8.0.0069 behaves like Neovim in this regard.
8.0.0250 seems to be the missing patch: https://github.com/vim/vim/commit/0c0590d9827cb07a33c1552cb3558b94bddcb4dc

**Comment by justinmk** [comment]
Ah, was just discussing this with @fmoralesc @jamessan @lunixbochs last week.

**Comment by jamessan** [comment]
Fixed by eb8cbfc.
